### DISCUSSION TITLE
Pulling image from AWS ECR private repository

### DISCUSSION THREAD

**Discussion by massimogallina** [kairos-io/kairos#1424] [Q&A]
Hello,
I would like to do a manual upgrade as described in the [guide](https://kairos.io/docs/upgrade/manual/) using ain image stored on a private repository on AWS ECR.
From what I understood I have to create a file like this:

```json
{
	"auths": {
		"aws_account_id.dkr.ecr.region.amazonaws.com": {
			"auth": "aws-token"
		}
	}
}
```

However, to generate the `aws-token` I need the AWS CLI installed on the host; could you confirm?
And is docker installed by default on Kairos images or should I install it before to run the upgrade command?

Thanks,
M

**Comment by jimmykarily** [comment]
I think you should generate the token outside Kairos and just put it there. Even if you had aws cli inside Kairos, you would need to verify your identity somehow in order to generate a token (using credentials or some other flow).
